src/App.js

```diff
@@ -1,7 +1,67 @@
+import { Route, Router, Routes } from "react-router-dom";
 import "./App.css";
 
+import {
+  SidebarLayoutPage,
+  Homepage,
+  Likepage,
+  Playlistpage,
+  Historypage,
+  WatchLaterpage,
+} from "./pages/index";
+
+import { useDispatch } from "react-redux";
+import { Login } from "./pages/Login/Login";
+import { SignUp } from "./pages/SignUp/SignUp";
+import { fetchVideos } from "./store/videoSlice";
+import { RequireAuth } from "./requireAuth";
+
 function App() {
-  return <div className="App"></div>;
+  const dispatch = useDispatch();
+  dispatch(fetchVideos());
+  return (
+    <div className="App">
+      <Routes>
+        <Route element={<SidebarLayoutPage />}>
+          <Route path="/" element={<Homepage />}></Route>
+          <Route
+            path="/likes"
+            element={
+              <RequireAuth>
```

Comment:
suggestion: Wrap all the private routes under ```RequireAuth``` instead of wrapping each route separately.